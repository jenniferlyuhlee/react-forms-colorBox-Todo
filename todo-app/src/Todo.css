.Todo-group{
    display: flex;
    margin: 10px 0;
}

.Todo-btn{
    margin-left: 5px;
    font-size: 0.5em;
    padding: 0 7px;
}

.Todo-btn:hover{
    background-color: rgba(255, 0, 0, 0.4);
}

.done-btn:hover{
    background-color: rgba(0, 255, 0, 0.4);
}

.task-done{
    text-decoration: line-through;
}

