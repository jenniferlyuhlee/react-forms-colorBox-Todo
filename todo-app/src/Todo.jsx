import React, { useState } from "react";
import "./Todo.css";

/**
 * Component Todo
 * Props: task
 */

function Todo({task, removeTodo}){
    const [isDone, setIsDone] = useState(false);

    function addStrikethrough(){
        setIsDone(!isDone);
    }

    return(
        <div className = "Todo-group">
            <li className={`Todo-task ${isDone ? "task-done" : ""}`}>
                {task}
            </li>
            <button className="Todo-btn" onClick = {removeTodo}>Delete</button>
            <button className="Todo-btn done-btn"
                    onClick = {addStrikethrough}>   
                    Mark as done
            </button>
        </div>
    )
}

export default Todo;