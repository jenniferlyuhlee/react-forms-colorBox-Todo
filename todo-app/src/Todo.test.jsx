import React from "react";
import {fireEvent, render} from "@testing-library/react"
import Todo from "./Todo";

// smoke test
test("it renders without crashing", () => {
    render (<Todo task = "test"/>)
})

// snapshot test
test("it matches snapshot", () => {
    const {asFragment} = render(<Todo task = "test" />)
    expect(asFragment()).toMatchSnapshot();
})

// specialized test
test("strikethrough toggling works", () => {
    const {getByText} = render(<Todo task = "test" />);
    const doneBtn = getByText("Mark as done")
    const todo = getByText("test")
    
    expect(todo).not.toHaveClass("task-done")

    fireEvent.click(doneBtn)

    expect(todo).toHaveClass("task-done")
})