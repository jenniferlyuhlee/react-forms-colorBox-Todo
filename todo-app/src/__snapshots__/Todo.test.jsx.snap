// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo-group"
  >
    <li
      class="Todo-task "
    >
      test
    </li>
    <button
      class="Todo-btn"
    >
      Delete
    </button>
    <button
      class="Todo-btn done-btn"
    >
      Mark as done
    </button>
  </div>
</DocumentFragment>
`;
